### PR TITLE
✨ RW-76: Trees map

### PR DESCRIPTION
**Pull request by tulustul**

### CODE REVIEW

**Comment by mateusz-sznura-ulam** [comment]
Also is this still only a POC or a production code already? Because you might want to update PR title if that's final code

File: marketplace-app/src/pages/Map/SelectedTree.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          >
+            <TreeDetails tree={treeDetails} />
+          </Modal>
+        )}
+      </>
+    );
+  }
+
+  return (
+    <div className='absolute top-10 right-10 bg-teal-100 dark:bg-teal-900 rounded-2xl p-5 z-[1000] flex flex-col'>
```
[the remark this answers is not in the document]
Comment (tulustul) [reply marketplace-app/src/pages/Map/SelectedTree.tsx:77]: The Leaflet uses z-index in hundreds. We must be higher than that.

File: marketplace-app/src/modules/api/base-api.ts [hunk 2]
```diff
@@ -11,7 +11,6 @@ export const api = axios.create({
   baseURL: apiBaseURL,
 });
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
[the remark this answers is not in the document]
Comment (tulustul) [reply marketplace-app/src/modules/api/base-api.ts:14]: I disabled the rule

File: marketplace-app/src/index.css [hunk 3]
```diff
@@ -25,3 +27,23 @@
...
+  width: 30px;
+  height: 30px;
+  box-shadow: 0 0 4px 0 #555;
+  color:black;
+}
+
+.leaflet-div-icon {
+  border: none;
+  background:none;
+}
```
[the remark this answers is not in the document]
Comment (tulustul) [reply marketplace-app/src/index.css:49]: I think I have to keep `leaftlet-div-icon` here.

* Tailwind - `leaftlet-div-icon` is an internal class of Leaflet which I'm overriding. From Tailwind perspective this class is not used and therefore removed from the production build. 
* CSS modules - adds a unique suffix to classes

**Comment by tulustul** [comment]
I will prepare a separate PR with a changed directory structure. I like the `modules` approach.

File: marketplace-app/src/pages/Map/SelectedTree.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mateusz-sznura-ulam) [reply marketplace-app/src/pages/Map/SelectedTree.tsx:77]: can you add a short comment with explanation?

File: marketplace-app/src/common/components/Modal.tsx [hunk 4]
```diff
@@ -11,7 +11,7 @@ type ModalProps = PropsWithChildren & {
 export function Modal(props: ModalProps) {
   const modal = (
     <div
-      className='w-full h-dvh flex items-center justify-center absolute top-0 left-0 bg-black bg-opacity-50 z-10'
+      className='w-full h-dvh flex items-center justify-center absolute top-0 left-0 bg-black bg-opacity-50 z-[2000]'
```
[the remark this answers is not in the document]
Comment (mateusz-sznura-ulam) [reply marketplace-app/src/common/components/Modal.tsx:14]: can you add a short comment with explanation?

**Review by mateusz-sznura-ulam — APPROVED** [review]

File: backend/replant/api/trees_cluster.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        fields = (
+            "id",
+            "latitude",
+            "longitude",
+            "number_of_trees",
+            "zoom",
+        )
+
+
+class TreesClustersView(generics.ListAPIView):
```
Comment (kasia-antos) [inline backend/replant/api/trees_cluster.py:19]: I would add an API doc here 

```python
from drf_spectacular.utils import extend_schema, extend_schema_view, OpenApiParameter


@extend_schema_view(
    get=extend_schema(
        parameters=[
            OpenApiParameter(name='zoom', type=int, required=True),
            OpenApiParameter(name='index', type=int, required=True),
        ]
    )
)
class TreesClustersView(generics.ListAPIView):
...

File: backend/replant/api/tree_point.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            "nft_id",
+            "latitude",
+            "longitude",
+        )
+
+    latitude = serializers.FloatField()
+    longitude = serializers.FloatField()
+
+
+class TreePointsView(generics.ListAPIView):
```
Comment (kasia-antos) [inline backend/replant/api/tree_point.py:20]: I would add an API doc here


```python
from drf_spectacular.utils import extend_schema, extend_schema_view, OpenApiParameter


@extend_schema_view(
    get=extend_schema(
        parameters=[
            OpenApiParameter(name='index', type=int, required=True),
        ]
    )
)
class TreesClustersView(generics.ListAPIView):
...

**Review by kasia-antos — APPROVED** [review]
Really nice! 🏅 🚀  One more thing for the future maybe we can store the data in some temporary table or at least update it at the very end of the process as right now if something goes wring during the computation or if the computation takes a lot of time the users will see incomplete or wrong data on the map.

**Pull request merged**